Marlin/src/module/temperature.h

```diff
@@ -961,9 +961,9 @@ class Temperature {
     #endif
 
     #if HAS_HOTEND && HAS_STATUS_MESSAGE
-      static void set_heating_message(const uint8_t e);
+      static void set_heating_message(const uint8_t e, const bool persist=false);
     #else
-      static void set_heating_message(const uint8_t) {}
+      static void set_heating_message(const uint8_t, const bool=false) {}
```

Comment:
For heating messages, it would be better to set a flag that Marlin should take note that the target temperature has been reached and then reset the "heating" message (if it was not overwritten), not just expire it at some arbitrary time.

Reply:
If there are many criteria whereby a status message might get reset for a future condition, then a `status_callback_fn` may be added that will return `true` when the status message ought to be reset. Then whenever the status message is set to something simple, the callback ptr will be cleared.